src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -341,51 +345,67 @@ static void read_channel_args(grpc_chttp2_transport* t,
-                       {true, false}},
-                      {GRPC_ARG_HTTP2_HPACK_TABLE_SIZE_DECODER,
-                       GRPC_CHTTP2_SETTINGS_HEADER_TABLE_SIZE,
-                       -1,
-                       0,
-                       INT32_MAX,
-                       {true, true}},
-                      {GRPC_ARG_MAX_METADATA_SIZE,
-                       GRPC_CHTTP2_SETTINGS_MAX_HEADER_LIST_SIZE,
-                       -1,
-                       0,
-                       INT32_MAX,
-                       {true, true}},
-                      {GRPC_ARG_HTTP2_MAX_FRAME_SIZE,
-                       GRPC_CHTTP2_SETTINGS_MAX_FRAME_SIZE,
-                       -1,
-                       16384,
-                       16777215,
-                       {true, true}},
-                      {GRPC_ARG_HTTP2_ENABLE_TRUE_BINARY,
-                       GRPC_CHTTP2_SETTINGS_GRPC_ALLOW_TRUE_BINARY_METADATA,
-                       1,
-                       0,
-                       1,
-                       {true, true}},
-                      {GRPC_ARG_HTTP2_STREAM_LOOKAHEAD_BYTES,
-                       GRPC_CHTTP2_SETTINGS_INITIAL_WINDOW_SIZE,
-                       -1,
-                       5,
-                       INT32_MAX,
-                       {true, true}}};
+  } settings_map[] = {
+      {GRPC_ARG_MAX_CONCURRENT_STREAMS,
+       GRPC_CHTTP2_SETTINGS_MAX_CONCURRENT_STREAMS,
+       -1,
+       0,
+       INT32_MAX,
+       {true, false}},
+      {GRPC_ARG_HTTP2_HPACK_TABLE_SIZE_DECODER,
+       GRPC_CHTTP2_SETTINGS_HEADER_TABLE_SIZE,
+       -1,
+       0,
+       INT32_MAX,
+       {true, true}},
+      {GRPC_ARG_MAX_METADATA_SIZE,
+       GRPC_CHTTP2_SETTINGS_MAX_HEADER_LIST_SIZE,
+       -1,
+       0,
+       INT32_MAX,
+       {true, true}},
+      {GRPC_ARG_HTTP2_MAX_FRAME_SIZE,
+       GRPC_CHTTP2_SETTINGS_MAX_FRAME_SIZE,
+       -1,
+       16384,
+       16777215,
+       {true, true}},
+      {GRPC_ARG_HTTP2_ENABLE_TRUE_BINARY,
+       GRPC_CHTTP2_SETTINGS_GRPC_ALLOW_TRUE_BINARY_METADATA,
+       1,
+       0,
+       1,
+       {true, true}},
+      {GRPC_ARG_HTTP2_STREAM_LOOKAHEAD_BYTES,
+       GRPC_CHTTP2_SETTINGS_INITIAL_WINDOW_SIZE,
+       -1,
+       5,
+       INT32_MAX,
+       {true, true}},
+      {GRPC_ARG_EXPERIMENTAL_HTTP2_PREFERRED_CRYPTO_FRAME_SIZE,
+       GRPC_CHTTP2_SETTINGS_GRPC_PREFERRED_RECEIVE_CRYPTO_FRAME_SIZE,
+       -1,
+       16384,
+       16777215,
+       {true, true}}};
 
   for (size_t i = 0; i < GPR_ARRAY_SIZE(settings_map); i++) {
     const auto& setting = settings_map[i];
     if (setting.availability[is_client]) {
       const int value = channel_args.GetInt(setting.channel_arg_name)
                             .value_or(setting.default_value);
       if (value >= 0) {
-        queue_setting_update(t, setting.setting_id,
-                             grpc_core::Clamp(value, setting.min, setting.max));
+        if (setting.setting_id !=
```

Comment:
I think this branch can be removed now: queue_setting_update performs no action if the flag is not at its default, and we shouldn't change the default if the relevant channel arg is not set right?

Reply:
Removed it now.